### DISCUSSION TITLE
By default, Vue router should support filesystem-based routing.

### DISCUSSION THREAD

**Discussion by yogeshgalav** [vuejs/router#1631] [Polls]
As per design of all new frameworks (NUXT, NEXT, SvelteKit) there is a common thing, filesystem-based routing.
Even though this feature can also be implemented with vite-plugin-pages, it will be more advantageous to include this feature in package itself.

Implementing this feature into vue-router package will give following advantages:
1.) Standarization of folder structure.
2.) Ease of development due to easy finding of files. 
3.) Reduction of code, time and effort used to write lot of routes.

I struggled lot of years to find standard way of folder structure in frontend till I came across filesystem-based routing. It wouldn't be the same case if vue-documentation or vue-router had told me to do so in beginning. Also, I would have saved so many hours and effort writing frontend routes. 

It's better to give developers flexibility but it's best to set some standards.

Above mentioned advantages will make Vue the next era framework.

**Comment by posva** [comment]
👀  https://github.com/posva/unplugin-vue-router/

**Reply by yogeshgalav** [reply]
Just need some mention in original docs so every beginner knows about it.
Standarization is the main thing that I seek.

**Comment by yogeshgalav** [comment]
@yyx990803 
Hello sir,
With this feature we could take Vue-Router and vue to next level.
Also, if we can directly pass meta information that would be more SEO friendly
```
$router.path('/posts').withMeta({
    title : 'abc'
    description : 'xyz'
}) ;
```

**Reply by posva** [reply]
https://github.com/unjs/unhead
